kedro/templates/project/hooks/post_gen_project.py

```diff
@@ -0,0 +1,114 @@
+import json
+from pathlib import Path
+import shutil
+
+
+def parse_add_ons_input(add_ons_str):
+    """Parse the add-ons input string.
+
+    Args:
+        add_ons_str: Input string from prompts.yml.
+
+    Returns:
+        list: List of selected add-ons as strings.
+    """
+    if add_ons_str == "all":
+        return ["1", "2", "3", "4", "5"]
+    if add_ons_str == "none":
+        return []
+
+    # Split by comma
+    add_ons_choices = add_ons_str.split(",")
+    selected = []
+
+    for choice in add_ons_choices:
+        if "-" in choice:
+            start, end = choice.split("-")
+            selected.extend(str(i) for i in range(int(start), int(end) + 1))
+        else:
+            selected.append(choice.strip())
+
+    return selected
+
+
+# Get the selected add-ons from cookiecutter
+selected_add_ons = "{{ cookiecutter.add_ons }}"
+
+# Parse the add-ons to get a list
+selected_add_ons_list = parse_add_ons_input(selected_add_ons)
+
+current_dir = Path.cwd()
+requirements_file_path = current_dir / "src/requirements.txt"
+pyproject_file_path = current_dir / "src/pyproject.toml"
+
+if "1" not in selected_add_ons_list:  # If Linting not selected
+    setup_cfg_path = current_dir / "setup.cfg"
+    if setup_cfg_path.exists():
+        setup_cfg_path.unlink()
+else:
+    with open(requirements_file_path, 'a') as file:
+        file.write(
+            "black~=22.0\nflake8>=3.7.9, <5.0\nisort~=5.0")
```

Comment:
For the linting requirements I think this would be a good place to switch to ruff (as mentioned in some of the comments above - all the necessary requirements have been detailed on [this documentation page](https://docs.kedro.org/en/latest/development/linting.html))

Reply:
Additionally there is some configuration that will need to go in the project level pyproject.toml, it's all listed under the documentation link 😄

Reply:
Done that, Did not add any specific version because the documentation doesn't ask for any, let me know if there's a specific version to be added.